### DISCUSSION TITLE
How would I slow the simulation down?

### DISCUSSION THREAD

**Discussion by IBuildVans** [mars-sim/mars-sim#1117] [Q&A] [answered]
I am running the headless version on a linux server.   With the default settings, about 40 sols passes for every hour of real time.   What would I need to change to get it closer to maybe 10 sols per hour?

**Comment by mokun** [comment, marked as answer]
> What would I need to change...

Yes. The *speed* of the simulation can be adjusted by clicking the `+` or `-` button on the bottom left of the main window. See below : 

![image](https://github.com/mars-sim/mars-sim/assets/1168584/aa82c336-b773-4ee5-86cc-1bc8e8756a36)

Also, you can activate the `Time Tool` using keyboard by hitting F3 

![image](https://github.com/mars-sim/mars-sim/assets/1168584/befe5229-fea2-46b0-9b7f-98682d117b30)

or by clicking on the *clock* icon on top menu bar.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/394037ef-2dfd-46f7-a9f8-2c4712accc45)

>    get it closer to maybe 10 sols per hour?

Unfortunately, we don't have a direct way to adjust the speed to your desired setting.

But the short answer is setting the *Time Ratio (TR)* to `256x`.

The detail calculation is as follows : 

As you can see below, my simulation session has Time Ratio (TR) set to 576x.

![image](https://github.com/mars-sim/mars-sim/assets/1168584/4ecbd8c2-fc97-4f43-8540-cad495afa92d)

Currently in that screenshot, one second of real time equals 576 sim seconds, or 9 minutes and 35 seconds. (or 09m 35s)
> 576 / 60 = 9.6 minutes
> 0.6 * 60 = 36 seconds

Given that, on earth, one real hour is 60 * 60 or 3600 real seconds.

But on Mars, one sol has 1000 millisols.

> [!NOTE]
> Real -> Earth
> Sim -> Mars 

By definition, One sol on Mars has 1000 millisols.

And one milisol on Mars is about 88.775244 Earth seconds. 

So, one sol on Mars is 88775.244 Earth seconds. 10 sols on Mar is approximately 887752.44 real seconds.

Now the time ratio (TR) you're interested in setting to  : 
> TR = 887752.44 / 3600 = 246.5979 ~ 246.6

On MasterClock, we define the following 3 legitimate sets of TRs : 
```
// low to mid :
// 1x, 2x, 4x, 8x, 16x, 32x, 64x, 128x, 256x 

// mid to high :
// 384x, 576x, 864x, 1296x, 1944x, 2916x

// high to max :
// 3645x, 7290x, 10935x, 14580x, 18225x, 21870x
```

So as soon as you use `+` or `-`  to adjust the TR to `256x`, it will give you 921600 sim seconds.
> 256 * 3600 = 921600

That's because 921600 sim seconds equals 10.381 sim sols.
> 256 * 3600 / 887752.44 = 10.381

Feel free to play around and let us know if you need any kind of help.

cc @bevans2000 @Urwumpe

**Comment by bevans2000** [comment]
With the headless version you will be logged into the console via a SSH terminal so use the 'speed' command to slow down or speed up.
Run 'help' for a list of commands.

**Reply by IBuildVans** [reply]
Thank You for the reply!   I did see the speed command but I didnt really understand what it did.   I am going to try turning it down to 256 and see how that works.

**Comment by IBuildVans** [comment]
Thank You for the info!   That is exactly what I needed.   I knew it had something to do with the Time Ratio but math was never my strong point.  On a side note,  I let the sim run over night and when I checked today many of them had sadly died.  Is that normal?   I think about a year of sim time had passed.  Thank You again.

**Reply by mokun** [reply]
I've made some adjustments on the time ratios to make the simulation a bit more stable in [this](https://github.com/mars-sim/mars-sim/commit/0632da6b0bcdf9edfadf24160e8dae78fcf5a5ed) and [that](https://github.com/mars-sim/mars-sim/commit/ac563fb6ff4ba21acaaaee04210cd91a3a82ef7e) commit last week ever since we started discussing this topic. 

You'll see it works a better in the upcoming 3.7.0 release. 

Do feel free to raise whatever questions you have in mind.

**Comment by mokun** [comment]
Hi @IBuildVans 

v3.7.0 has just been released. Sorry for the delay. 

See if you like better.

Also feel free to comment and contribute !

**Comment by IBuildVans** [comment]
Thanks, I'll check it out!
